Build failure: libsForQt5.qt5.qtwebengine on aarch64-darwin

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `libsForQt5.qt5.qtwebengine` on aarch64-darwin

### Build log

```cpp
FAILED: obj/media/ffmpeg/ffmpeg/ffmpeg_common.o 
/nix/store/sdhwc9qg1fi8m9l91idjb62x5ral2zvx-clang-wrapper-15.0.7/bin/clang++ -MMD -MF obj/media/ffmpeg/ffmpeg/ffmpeg_common.o.d -DOFFICIAL_BUILD -DTOOLKIT_QT -D_LIBCPP_HAS_NO_ALIGNED_ALLOCATION -DCR_XCODE_VERSION=0941 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DIS_MEDIA_IMPL -DUSE_NEON -DWEBP_EXTERN=extern -DUSE_EGL -DOPENSSL_NO_ASM -DSK_CODEC_DECODES_PNG -DSK_CODEC_DECODES_WEBP -DSK_ENCODE_PNG -DSK_ENCODE_WEBP -DSK_USER_CONFIG_HEADER=\"../../skia/config/SkUserConfig.h\" -DSK_GL -DSK_CODEC_DECODES_JPEG -DSK_ENCODE_JPEG -DSK_USE_LIBGIFCODEC -DSK_SUPPORT_GPU=1 -DSK_GPU_WORKAROUNDS_HEADER=\"gpu/config/gpu_driver_bug_workaround_autogen.h\" -DSK_BUILD_FOR_MAC -DSK_METAL -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DUCHAR_TYPE=uint16_t -Igen -I../../3rdparty/chromium -I../../3rdparty/chromium/third_party/perfetto/include -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -I../../3rdparty/chromium/third_party/libwebp/src -I../../3rdparty/chromium/third_party/libyuv/include -I../../3rdparty/chromium/third_party/khronos -I../../3rdparty/chromium/gpu -I../../3rdparty/chromium/third_party/abseil-cpp -I../../3rdparty/chromium/third_party/boringssl/src/include -I../../3rdparty/chromium/third_party/protobuf/src -Igen/protoc_out -I../../3rdparty/chromium/third_party/skia -I../../3rdparty/chromium/third_party/libgifcodec -I../../3rdparty/chromium/third_party/mesa_headers -I../../3rdparty/chromium/third_party/icu/source/common -I../../3rdparty/chromium/third_party/icu/source/i18n -I../../3rdparty/chromium/third_party/ffmpeg/chromium/config/Chrome/mac/arm64 -I../../3rdparty/chromium/third_party/ffmpeg -fno-strict-aliasing -fstack-protector -Wno-unknown-attributes -Wno-unknown-pragmas -Wno-unknown-warning-option -fcolor-diagnostics -fmerge-all-constants -arch arm64 -mcpu=apple-a12 -Wall -Wextra -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Wimplicit-fallthrough -Wunreachable-code -Wthread-safety -Wextra-semi -Wunguarded-availability -Wno-missing-field-initializers -Wno-unused-parameter -Wno-c++11-narrowing -Wno-unneeded-internal-declaration -Wno-undefined-var-template -Wno-psabi -Wno-ignored-pragma-optimize -Wno-implicit-int-float-conversion -Wno-final-dtor-non-final-class -Wno-builtin-assume-aligned-alignment -Wno-deprecated-copy -Wno-non-c-typedef-for-linkage -Wno-max-tokens -O2 -fno-omit-frame-pointer -g0 -isysroot ../../../../../../../nix/store/60n185lykh89xmvmjl4nd6difsfa4bi3-xcodebuild-0.1.2-pre/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk -mmacosx-version-min=10.13 -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Wno-shorten-64-to-32 -Wexit-time-destructors -Wno-parentheses-equality -Wno-tautological-compare -Wno-thread-safety-attributes -Wno-undefined-bool-conversion -Wno-tautological-undefined-compare -std=c++14 -fno-trigraphs -Wno-trigraphs -stdlib=libc++ -fno-exceptions -fno-rtti -fvisibility-inlines-hidden -c ../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.cc -o obj/media/ffmpeg/ffmpeg/ffmpeg_common.o
In file included from ../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.cc:5:
In file included from ../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.h:15:
In file included from ../../3rdparty/chromium/base/time/time.h:61:
In file included from ../../3rdparty/chromium/base/check_op.h:12:
../../3rdparty/chromium/base/template_util.h:138:7: warning: builtin __has_trivial_copy is deprecated; use __is_trivially_copyable instead [-Wdeprecated-builtins]
      __has_trivial_copy(T) && __has_trivial_destructor(T);
      ^
../../3rdparty/chromium/base/template_util.h:138:32: warning: builtin __has_trivial_destructor is deprecated; use __is_trivially_destructible instead [-Wdeprecated-builtins]
      __has_trivial_copy(T) && __has_trivial_destructor(T);
                               ^
../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.cc:344:22: error: no member named 'ch_layout' in 'AVCodecContext'
      codec_context->ch_layout.nb_channels > 8
      ~~~~~~~~~~~~~  ^
../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.cc:347:32: error: no member named 'ch_layout' in 'AVCodecContext'
                codec_context->ch_layout.u.mask,
                ~~~~~~~~~~~~~  ^
../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.cc:348:32: error: no member named 'ch_layout' in 'AVCodecContext'
                codec_context->ch_layout.nb_channels);
                ~~~~~~~~~~~~~  ^
../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.cc:401:51: error: no member named 'ch_layout' in 'AVCodecContext'
    config->SetChannelsForDiscrete(codec_context->ch_layout.nb_channels);
                                   ~~~~~~~~~~~~~  ^
../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.cc:418:26: warning: '&&' within '||' [-Wlogical-op-parentheses]
  if (codec == kCodecAAC && codec_context->profile == FF_PROFILE_UNKNOWN ||
      ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~
../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.cc:418:26: note: place parentheses around the '&&' expression to silence this warning
  if (codec == kCodecAAC && codec_context->profile == FF_PROFILE_UNKNOWN ||
      ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../3rdparty/chromium/media/ffmpeg/ffmpeg_common.cc:468:18: error: no member named 'ch_layout' in 'AVCodecContext'
  codec_context->ch_layout.nb_channels = config.channels();
  ~~~~~~~~~~~~~  ^
```


### Notify maintainers

@matthewbauer 
also @emilazy since #330648 was merged not so long ago, it might be the issue, I'm checking this

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.5.0, macOS 14.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Lix, like Nix) 2.90.0-rc1`
 - nixpkgs: `/nix/store/f5ykynf933aab7nd1arqz6dm79hdh91q-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/qt-5/modules/qtwebengine.nix
+++ b/pkgs/development/libraries/qt-5/modules/qtwebengine.nix
@@ -173,6 +173,10 @@ qtModule ({
     # ld: fatal warning(s) induced error (-fatal_warnings)
     substituteInPlace src/3rdparty/chromium/build/config/compiler/BUILD.gn \
       --replace "-Wl,-fatal_warnings" ""
+
+    # Use system ffmpeg
+    echo "gn_args += use_system_ffmpeg=true" >> src/core/config/mac_osx.pri
+    echo "LIBS += -lavformat -lavcodec -lavutil" >> src/core/core_common.pri
   '') + postPatch;
 
   env = {

